process/smartContract/scQueryService.go

```diff
@@ -167,104 +169,111 @@ func (service *SCQueryService) shouldAllowQueriesExecution() bool {
 	}
 }
 
-func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice uint64) (*vmcommon.VMOutput, error) {
+func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice uint64) (*vmcommon.VMOutput, common.BlockInfo, error) {
 	log.Trace("executeScCall", "function", query.FuncName, "numQueries", service.numQueries)
 	service.numQueries++
 
 	shouldEarlyExitBecauseOfSyncState := query.ShouldBeSynced && service.bootstrapper.GetNodeState() == common.NsNotSynchronized
 	if shouldEarlyExitBecauseOfSyncState {
-		return nil, process.ErrNodeIsNotSynced
+		return nil, nil, process.ErrNodeIsNotSynced
+	}
+
+	blockHeader, blockRootHash, err := service.extractBlockHeaderAndRootHash(query)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var blockHash []byte
+	var blockNonce uint64
+
+	if !check.IfNil(blockHeader) {
+		blockNonce = blockHeader.GetNonce()
+		blockHash, err = service.getBlockHashByNonce(blockNonce)
```

Comment:
you don't need storage access just for fetching the hash. it's better to simply compute the hash of the already existing header.
also, these lines can be moved before the end of function, since blockHash and blockNonce are only used for populating the `blockInfo`

Reply:
updated as suggested